### ISSUE TITLE
why all the loaded data are deleted after the command "ycsb run elasticsearch"

### ISSUE THREAD

**Issue by lukawendy** [brianfrankcooper/YCSB#796]
I have already run "ycsb load elasticsearch" successfully, but then after I run "ycsb run elasticsearch", the records are all deleted. I have to load again, then run again. Is there a way that I can "ycsb run elasticsearch" with different workloads several times and get the results?

**Comment by busbey** [comment]
The dataset should persist across runs. Let's see if maybe @jasontedor knows what's up.

**Comment by jasontedor** [comment]
@lukawendy I suspect that this is because you're running in embedded mode with the defaults.

The Elasticsearch binding has two modes of operation, embedded mode and remote mode. In embedded mode, the client creates an embedded instance of Elasticsearch that uses a temporary data path which will not be the same between runs. However, if you specify the same data path between runs via `-p path.data=<path>` then the data will persist between runs.

For example, you could do:

```bash
$ mkdir /tmp/es
$ ./bin/ycsb.sh load elasticsearch -P workloads/workloada -p path.home=/tmp/es
$ ./bin/ycsb.sh run elasticsearch -P workloads/workloada -p path.home=/tmp/es
```

and the data will be persisted across runs.

In remote mode, the client will hit a standalone installation of Elasticsearch which could be a single-node instance on the same machine, or a remote cluster.

For example, you could start Elasticsearch:

```bash
$ <elasticsearch-path>/bin/elasticsearch --cluster.name=es.ycsb.cluster -d
$ curl -XGET localhost:9200/
{
  "name" : "Battletide",
  "cluster_name" : "es.ycsb.cluster",
  "version" : {
    "number" : "2.3.3",
    "build_hash" : "218bdf10790eef486ff2c41a3df5cfa32dadcfde",
    "build_timestamp" : "2016-05-17T15:40:04Z",
    "build_snapshot" : false,
    "lucene_version" : "5.5.0"
  },
  "tagline" : "You Know, for Search"
}
$ ./bin/ycsb.sh load elasticsearch -P workloads/workloada -p es.remote=true
$ .bin/ycsb.sh run elasticsearch -P workloads/workloada -p es.remote=true
```

If you want to connect to a cluster on another machine, you can use a comma separated list of hostname/transport port pairs: `-p es.hosts=<hostname1:port1>,...,<hostname2:port2>`.

**Comment by lukawendy** [comment]
Yes, that's it! Thank you soooo much! @busbey  @jasontedor

**Comment by busbey** [comment]
That's a great explanation @jasontedor. Could we add it to the README for the elasticsearch module?

**Comment by jasontedor** [comment]
@busbey Sure, I'll open a PR later today.

**Comment by jasontedor** [comment]
@busbey It took me longer to get to than I had intended, but I opened #800. :smile:

**Comment by busbey** [comment]
closed by #800.
